### PR TITLE
Make backpacks more consistent

### LINKED ISSUE #33292: New Backpacks Need Adjusting
**Is your feature request related to a problem? Please describe.**  
New Backpacks were added.  They need balanced.

**Describe the solution you'd like**  
The new backpacks need some tweaking.  All have material_thickness of 6 which gives some of them pretty ridiculous protection values (compared to other storage items and their recipe cost)

They also are all auto-learned recipes, and I believe some should be locked into books.  

Similarly, the recipes could probably use some looking at.  For instance, the Travelpack requires 5 rags and gives 25 liters of storage.  Compared to the backpack, which requires 20 rags and gives 10 liters of storage.  These recipes need to be balanced against the existing recipes in-game, or, alternatively, the old items need to be balanced against the new items.  

Additionally, I believe their coverage should be looked at.  Specifically, the high-volume rucksack has 95% coverage.  Combining this with its 24 bash and 24 cut protection, this is quite an item.  I understand it gives 100 encumbrance, but it's still too much, in my opinion.

**Describe alternatives you've considered**  
None.  They need to be balanced against existing storage items or, alternatively, the old storage items need to be balanced against the new items.

**Additional context**  
I hate the new packs.  They're OP.  I'm uncertain of whether I've been objective in this request, so if I wasn't, I apologize.  I thought about changing them myself, but I wasn't sure how to go about this.

### PR DESCRIPTION
**Pull request by ampersand55**
#### Summary

```SUMMARY: Balance "Make backpacks more consistent```

#### Purpose of change

Adjust the new backpacks to be consistent with the old backpacks. 

Fixes #33292 (to some degree).

#### Describe the solution

We are given storage capacity. We assume that storage capacity is proportional to volume and material use and weight is proportional to surface area times material thickness. 

Assuming a backpack has the shape of a 1x1x2 box, it will have a surface area proportional to ~6.2996 x volume.

We normalize material use from to the regular backpack, which use 1 rag per storage per thickness (20 rags, 10L storage, 2 thickness). 

* Relative material cost of materials is calculated using the duffel bag recipe, which uses 40 rags, 32 fur pelts, 24 leather or 6 hides. 

* Old backpacks material is left as is.

* Warmth is calculated as coverage x material_thickness / 10. 

* Survivor gear warmth is left as is (at 0).

* Coverage of new backpacks is reduced. Assuming the high-volume rucksack covers the entire back it has a coverage of 50%.

* Other values are left as is.

* Removed "steel" as a material for high volume rucksack as it made its armor value far to high.

* Backpacks (all belted torso items used for general storage) moved to backpacks.json.

#### Describe alternatives you've considered

Backpacks probably need more rebalancing. Some things to consider for a future PR:

* Allow new backpacks to be made out of rags, and change default material to cotton.
* Take a look at weight and volume for the backpacks.
* Make survivor versions of new backpacks.

#### Additional context

https://docs.google.com/spreadsheets/d/1i72c0SF2o6ZMDmcTU2I4NBQKpshlr4Ps0y7cGH7Saco/edit?usp=sharing
https://en.wikipedia.org/wiki/Surface-area-to-volume_ratio

### CODE REVIEW

**Review by anothersimulacrum — COMMENTED** [review]
Moving items and making changes to the in the same PR is confusing, because it obfuscates what changes you make. Doing so in the same commit makes it incredibly impractical to tell what changes you made.

**Comment by ampersand55** [comment]
> 
> 
> Moving items and making changes to the in the same PR is confusing, because it obfuscates what changes you make. Doing so in the same commit makes it incredibly impractical to tell what changes you made.

That's true, sorry. I thought of it, but too late. 

The changes in backpack.json are to material, material_thickness, coverage and warmth for the new backpacks and warmth for the old backpacks.

I've detailed all changes I made in the spreadsheet. 

Should I make a new PR?

**Comment by anothersimulacrum** [comment]
No, the spreadsheet is fine, just please don't do it in the future.

File: data/json/items/armor/backpacks.json [hunk 1]
```diff
@@ -0,0 +1,482 @@
+[
+  {
+    "id": "backpack",
+    "type": "ARMOR",
+    "name": "backpack",
+    "description": "A small backpack.  Good storage for a little encumbrance.",
+    "weight": 633,
+    "volume": 8,
```
Comment (anothersimulacrum) [inline data/json/items/armor/backpacks.json]:
```suggestion
    "volume": "2 L",
```

File: data/json/items/armor/backpacks.json [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+    },
+    "flags": [ "BELTED", "WATERPROOF", "ONLY_ONE", "OVERSIZE" ]
+  },
+  {
+    "id": "backpack_giant",
+    "type": "ARMOR",
+    "name": "giant novelty backpack",
+    "description": "A huge fabric backpack made mostly as a joke before the cataclysm.  Now, it's still rather silly, but it can store a lot of stuff.",
+    "weight": 1200,
+    "volume": 35,
```
Comment (anothersimulacrum) [inline data/json/items/armor/backpacks.json]:
```suggestion
    "volume": "8750 ml",
```

File: data/json/items/armor/backpacks.json [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+    "material_thickness": 2,
+    "flags": [ "BELTED" ]
+  },
+  {
+    "id": "backpack_leather",
+    "type": "ARMOR",
+    "name": "leather backpack",
+    "description": "A small leather backpack.  Good storage for a little encumbrance.",
+    "weight": 818,
+    "volume": 8,
```
Comment (anothersimulacrum) [inline data/json/items/armor/backpacks.json]:
```suggestion
    "volume": "2 L",
```

File: data/json/items/armor/backpacks.json [hunk 4]
```diff
@@ -0,0 +1,482 @@
...
+    "material_thickness": 2,
+    "flags": [ "BELTED", "WATERPROOF", "ONLY_ONE", "OVERSIZE" ]
+  },
+  {
+    "id": "dive_bag",
+    "type": "ARMOR",
+    "name": "dive bag",
+    "description": "A lightweight mesh backpack, commonly worn by swimmers and divers.",
+    "weight": 512,
+    "volume": 6,
```
Comment (anothersimulacrum) [inline data/json/items/armor/backpacks.json]:
```suggestion
    "volume": "1500 ml",
```

File: data/json/items/armor/backpacks.json [hunk 5]
```diff
@@ -0,0 +1,482 @@
...
+    "material_thickness": 1,
+    "flags": [ "WATER_FRIENDLY", "BELTED" ]
+  },
+  {
+    "id": "duffelbag",
+    "type": "ARMOR",
+    "name": "duffel bag",
+    "description": "A huge duffel bag.  Provides plenty of storage, but is severely encumbering.",
+    "weight": 933,
+    "volume": 24,
```
Comment (anothersimulacrum) [inline data/json/items/armor/backpacks.json]:
```suggestion
    "volume": "6 L",
```

File: data/json/items/armor/backpacks.json [hunk 6]
```diff
@@ -0,0 +1,482 @@
...
+    "use_action": "CAPTURE_MONSTER_ACT",
+    "flags": [ "BELTED", "WATERPROOF" ]
+  },
+  {
+    "id": "jerrypack",
+    "type": "ARMOR",
+    "name": "jerrypack",
+    "description": "A jerrycan modified to be worn in a similar manner to a backpack.",
+    "weight": 1763,
+    "volume": 40,
```
Comment (anothersimulacrum) [inline data/json/items/armor/backpacks.json]:
```suggestion
    "volume": "10 L",
```

File: data/json/items/armor/backpacks.json [hunk 7]
```diff
@@ -0,0 +1,482 @@
...
+    "container_data": { "contains": 40, "seals": true, "watertight": true },
+    "flags": [ "OVERSIZE", "BELTED", "WATER_FRIENDLY" ]
+  },
+  {
+    "id": "makeshift_knapsack",
+    "type": "ARMOR",
+    "name": "makeshift knapsack",
+    "description": "A pair of pants with the bottom ends tied with string, and then passed through the belt loops so the knapsack can be carried on the back.",
+    "weight": 560,
+    "volume": 8,
```
Comment (anothersimulacrum) [inline data/json/items/armor/backpacks.json]:
```suggestion
    "volume": "2 L",
```

File: data/json/items/armor/backpacks.json [hunk 8]
```diff
@@ -0,0 +1,482 @@
...
+    "material_thickness": 1,
+    "flags": [ "BELTED", "WATER_FRIENDLY" ]
+  },
+  {
+    "id": "makeshift_sling",
+    "type": "ARMOR",
+    "name": "makeshift sling",
+    "description": "A large sheet tied into a crude, over-the-shoulder sling.",
+    "weight": 771,
+    "volume": 20,
```
Comment (anothersimulacrum) [inline data/json/items/armor/backpacks.json]:
```suggestion
    "volume": "5 L",
```

File: data/json/items/armor/backpacks.json [hunk 9]
```diff
@@ -0,0 +1,482 @@
...
+    "flags": [ "OVERSIZE", "BELTED", "WATER_FRIENDLY" ]
+  },
+  {
+    "id": "mbag",
+    "type": "ARMOR",
+    "name": "messenger bag",
+    "//": "KA101's ran about $90 but is ballistic nylon.  Bit tougher than the DDA model.",
+    "description": "Light and easy to wear, but doesn't offer much storage.",
+    "weight": 760,
+    "volume": 4,
```
Comment (anothersimulacrum) [inline data/json/items/armor/backpacks.json]:
```suggestion
    "volume": "1 L",
```

File: data/json/items/armor/backpacks.json [hunk 10]
```diff
@@ -0,0 +1,482 @@
...
+    "material_thickness": 1,
+    "flags": [ "BELTED", "WATER_FRIENDLY" ]
+  },
+  {
+    "id": "molle_pack",
+    "type": "ARMOR",
+    "name": "MOLLE pack",
+    "description": "The Modular Lightweight Load-carrying Equipment is an advanced military backpack.  Covered with pockets and straps, it strikes a fine balance between storage space and encumbrance.",
+    "weight": 966,
+    "volume": 10,
```
Comment (anothersimulacrum) [inline data/json/items/armor/backpacks.json]:
```suggestion
    "volume": "2500 ml",
```

File: data/json/items/armor/backpacks.json [hunk 11]
```diff
@@ -0,0 +1,482 @@
...
+    "material_thickness": 2,
+    "flags": [ "BELTED", "WATER_FRIENDLY" ]
+  },
+  {
+    "id": "purse",
+    "type": "ARMOR",
+    "name": "purse",
+    "description": "A bit cumbersome to wear, but provides some storage.",
+    "weight": 297,
+    "volume": 4,
```
Comment (anothersimulacrum) [inline data/json/items/armor/backpacks.json]:
```suggestion
    "volume": "1 L",
```

File: data/json/items/armor/backpacks.json [hunk 12]
```diff
@@ -0,0 +1,482 @@
...
+    "material_thickness": 2,
+    "flags": [ "FANCY", "BELTED", "WATER_FRIENDLY" ]
+  },
+  {
+    "id": "rucksack",
+    "type": "ARMOR",
+    "name": "military rucksack",
+    "description": "A huge military rucksack, provides a lot of storage.",
+    "weight": 1140,
+    "volume": 16,
```
Comment (anothersimulacrum) [inline data/json/items/armor/backpacks.json]:
```suggestion
    "volume": "4 L",
```

**Comment by ampersand55** [comment]
> 
> 
> No, the spreadsheet is fine, just please don't do it in the future.

Will do. I mean not do that in the future.

File: data/json/items/armor/backpacks.json [hunk 13]
```diff
@@ -0,0 +1,482 @@
...
+    "material_thickness": 2,
+    "flags": [ "BELTED", "WATER_FRIENDLY" ]
+  },
+  {
+    "id": "runner_bag",
+    "type": "ARMOR",
+    "name": "runner pack",
+    "description": "The obvious choice for outdoor athletes, this ergonomic backpack is light and easy to wear, ensuring comfort when carrying heavy and bulky items.",
+    "weight": 840,
+    "volume": 6,
```
Comment (anothersimulacrum) [inline data/json/items/armor/backpacks.json]:
```suggestion
    "volume": "1500 ml",
```

File: data/json/items/armor/backpacks.json [hunk 14]
```diff
@@ -0,0 +1,482 @@
...
+    "flags": [ "BELTED", "WATER_FRIENDLY" ]
+  },
+  {
+    "id": "slingpack",
+    "type": "ARMOR",
+    "name": "sling pack",
+    "//": "Depends on if you're getting a cheap school version or something more like a range bag.",
+    "description": "A simple single-sling backpack.  Easier to access than a normal backpack, but can't comfortably hold as much.",
+    "weight": 566,
+    "volume": 6,
```
Comment (anothersimulacrum) [inline data/json/items/armor/backpacks.json]:
```suggestion
    "volume": "1500 ml",
```

File: data/json/items/armor/backpacks.json [hunk 15]
```diff
@@ -0,0 +1,482 @@
...
+    "material_thickness": 1,
+    "flags": [ "BELTED", "WATER_FRIENDLY" ]
+  },
+  {
+    "id": "survivor_duffel_bag",
+    "type": "ARMOR",
+    "name": "survivor duffel bag",
+    "description": "A custom-built heavy duffel bag.  Durable and carefully crafted to hold as much stuff as possible.",
+    "weight": 1000,
+    "volume": 30,
```
Comment (anothersimulacrum) [inline data/json/items/armor/backpacks.json]:
```suggestion
    "volume": "7500 ml",
```

File: data/json/items/armor/backpacks.json [hunk 16]
```diff
@@ -0,0 +1,482 @@
...
+    "material_thickness": 2,
+    "flags": [ "WATER_FRIENDLY", "STURDY", "BELTED", "OVERSIZE" ]
+  },
+  {
+    "id": "survivor_pack",
+    "type": "ARMOR",
+    "name": "survivor backpack",
+    "description": "A custom-built backpack.  Durable and carefully crafted to hold as much stuff as possible.",
+    "weight": 600,
+    "volume": 10,
```
Comment (anothersimulacrum) [inline data/json/items/armor/backpacks.json]:
```suggestion
    "volume": "2500 ml",
```

File: data/json/items/armor/backpacks.json [hunk 17]
```diff
@@ -0,0 +1,482 @@
...
+    "material_thickness": 2,
+    "flags": [ "WATER_FRIENDLY", "STURDY", "BELTED" ]
+  },
+  {
+    "id": "survivor_rucksack",
+    "type": "ARMOR",
+    "name": "survivor rucksack",
+    "description": "A custom-built heavy backpack.  Durable and carefully crafted to hold as much stuff as possible.",
+    "weight": 800,
+    "volume": 20,
```
Comment (anothersimulacrum) [inline data/json/items/armor/backpacks.json]:
```suggestion
    "volume": "5 L",
```

File: data/json/items/armor/backpacks.json [hunk 18]
```diff
@@ -0,0 +1,482 @@
...
+    "material_thickness": 2,
+    "flags": [ "WATER_FRIENDLY", "STURDY", "BELTED", "OVERSIZE" ]
+  },
+  {
+    "id": "survivor_runner_pack",
+    "type": "ARMOR",
+    "name": "survivor runner pack",
+    "description": "A custom-built lightweight runner pack.  Durable and carefully crafted to hold as much stuff as possible.",
+    "weight": 440,
+    "volume": 7,
```
Comment (anothersimulacrum) [inline data/json/items/armor/backpacks.json]:
```suggestion
    "volume": "1750 ml",
```

File: data/json/items/armor/backpacks.json [hunk 19]
```diff
@@ -0,0 +1,482 @@
...
+    "material_thickness": 2,
+    "flags": [ "WATER_FRIENDLY", "STURDY", "BELTED" ]
+  },
+  {
+    "id": "golf_bag",
+    "type": "ARMOR",
+    "name": "golf bag",
+    "description": "A tall canvas and plastic bag with fold-out legs used for golfing.  It even has straps to be worn on the back.",
+    "weight": 1900,
+    "volume": 60,
```
Comment (anothersimulacrum) [inline data/json/items/armor/backpacks.json]:
```suggestion
    "volume": "15 L",
```

**Comment by ampersand55** [comment]
I could rather quickly make a json parser and change volume to be in liters and milliliters in all json files. Is that something we would want?

**Comment by anothersimulacrum** [comment]
Yes, just waiting on someone to do it. We have `adjust_values.py` that could be adjusted a bit to do that, or so I've heard.

**Comment by ampersand55** [comment]
Ok, it wasn't quite as trivial as I thought as music instruments use the key "volume" to denote sound volume and items that use proportional volume gets messed up.

**Comment by anothersimulacrum** [comment]
Please add the script to `./tools/` when you have it working!

**Comment by ampersand55** [comment]
> 
> 
> Please add the script to `./tools/` when you have it working!

Well. It's working but it's ugly and not robust.

**Comment by JacobKessler** [comment]
I like backpack consistency. Some thoughts:
 - Should the volume be based on the volume with the packmule trait? It seems like having the manufacturer claimed volume match up with a maximally-efficient survivor seems better than some survivors be able to get 40% more volume out of nowhere.
 -  Given the new realistic volumes, 1L/encumbrance seems excessive. A hiking pack is not as encumbering as a poorly-fitting entry suit or wrapping yourself in a blanket. In fact, it seems like the more-advanced packs might want to give better storage/encumbrance ratios at the expense of weight due to the hip and back supports, like the vanilla MOLLE pack. Duffel bags seems like the 'huge storage, very encumbering' class of storage containers.

**Comment by ampersand55** [comment]
I think rebalancing storage / encumbrance is beyond the scope of this PR, but I agree backpacks need to be looked at further.

**Comment by ampersand55** [comment]
I did a hard reset and made the balance changes in a separate commit.

**Pull request merged**